src/ui/StartUp.java

```diff
@@ -46,6 +50,39 @@ private StartUp(int numberOfColumns, int gridDimension, int gridMargin) {
         setPreferredSize(new Dimension(gridDimension, imageSize * numberOfRows + 2 * gridMargin));
         setBackground(Color.WHITE);
 
+        addMouseListener(new MouseAdapter() {
+            @Override
+            public void mousePressed(MouseEvent e) {
+                // used to let users to interact on the grid by clicking
+                // it's time to implement interaction with users to move tiles to solve the game !
+
+                // get position of the click
+                int ex = e.getX() - gridMargin;
+                int ey = e.getY() - gridMargin;
+
+                // click in the grid ?
+                if (ex < 0 || ex > gridSize || ey < 0 || ey > gridSize)
+                    return;
+
+                // get position in the grid
+                int c1 = ex / imageSize;
+                int r1 = ey / imageSize;
+
+                // we convert in the 1D coord
+                int clickPosition = r1 * numberOfColumns + c1;
+
+                // get position of the blank cell
+                int c2 = positivePosition % numberOfColumns;
+                int r2 = positivePosition / numberOfColumns;
+
+                if (clickPosition == positivePosition) {
+                    startProgram();
```

Comment:
`startProgram()` apare si pe linia **86**.

Reply:
Da. apelul din interiorul metodei mousePressed are rolul de restart dupa ce apesi pe imaginea corecta, iar al doilea e apelul principal prin care incepe programul. Eventual, ca sa fie mai explicit as putea defini o metoda restart() care sa implementeze in interiorul ei metoda startProgram() ?

Reply:
Da, poti face astfel.